src/utils/bond.hpp

```diff
@@ -0,0 +1,54 @@
+// Copyright 2023 Memgraph Ltd.
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.txt; by using this file, you agree to be bound by the terms of the Business Source
+// License, and you may not use this file except in compliance with the Business Source License.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0, included in the file
+// licenses/APL.txt.
+
+#include <cstddef>
+#include <memory>
+#include "storage/v2/delta.hpp"
+#include "utils/memory.hpp"
+#include "utils/pmr/list.hpp"
+
+template <typename Container>
+struct Bond {
+  explicit Bond(std::size_t initial_size)
+      : res_(std::make_unique<resource>(initial_size)),
+        container_(memgraph::utils::Allocator<Container>(res_.get()).template new_object<Container>()){};
```

Comment:
Why allocate the container inside the resource and store a pointer? Why not just have the container be passed the polymorphic_allocator (implicitly constructed by `memory_resource*`). And store the container as a member of Bond, then the container wouldn't be behind a pointer (simplifying move).
```suggestion
  explicit Bond(std::size_t initial_size)
      : res_(std::make_unique<resource>(initial_size)),
        container_(res_.get()){};
```